internal/proto/scan_test.go

```diff
@@ -46,3 +50,37 @@ var _ = Describe("ScanSlice", func() {
 		}))
 	})
 })
+
+func TestScan(t *testing.T) {
+	t.Parallel()
+
+	t.Run("time", func(t *testing.T) {
+		t.Parallel()
+
+		toWrite := time.Now()
+		var toScan time.Time
+
+		buf := new(bytes.Buffer)
+		wr := NewWriter(buf)
+		err := wr.WriteArg(toWrite)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		r := NewReader(buf)
+		b, err := r.readTmpBytesReply()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = Scan(b, &toScan)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !toWrite.Equal(toScan) {
+			t.Fatal(errors.New("toWrite and toScan is not equal"))
+		}
+	})
+
+}
```

Comment:
This is a rather complicated. Instead you could add a test like:

```
tm := time.Now()
rdb.Set(ctx, "foo", tm, 0)

var tm2 time.Time
rdb.Get(ctx, "foo").Scan(&tm2)

// check tm2 equals tm
```

Reply:
I had to move package to proto_test. Otherwise I had circular imports.